### DISCUSSION TITLE
Svelte support for TanStack Router

### DISCUSSION THREAD

**Discussion by NightMachinery** [TanStack/router#281] [Q&A]
Can this project be used without using react? I am looking for a client-side router for my svelte frontends ...

**Comment by tannerlinsley** [comment]
Give me a month and maybe?

Tanner Linsley

**Reply by cassepipe** [reply]
Has this been deprioritzed because of SvelteKit ?

Still hyped about it ! Hope it sees the light of day <3

**Reply by codewithcheese** [reply]
Love svelte, dislike Sveltekit routing. Tanstack Router is awesome. Tanstack Svelte Router please please 😁

**Reply by tannerlinsley** [reply]
It’s a ways off.

**Reply by algoflows** [reply]
Hows it looking now matey? @tannerlinsley

**Reply by schiller-manuel** [reply]
we don't have svelte but we have solid :)
so adding svelte as a third framework is much easier now since the foundation exists. do you want to contribute to that?

**Reply by algoflows** [reply]
Don't have the time at the moment personally. I am already working overtime at a fintech startup. That being said I'd use tanstack router in Svelte over any other routing solution. I've been hoping that support would be added for a quite a while now!

**Reply by pythonhubdev** [reply]
Any update on this would be great if we have Tanstack Router and Tanstack Start for Svelte. @tannerlinsley

**Reply by birkskyum** [reply]
> Any update on this would be great if we have Tanstack Router and Tanstack Start for Svelte.

one positive development in this direction is that tanstack svelte query recently added support for Svelte 5
- https://github.com/TanStack/query/discussions/7413#discussioncomment-14552366

**Comment by Prajwalg19** [comment]
Guys any chance this will pace up now as vue also got it's support ?

**Reply by birkskyum** [reply]
It was helpful when adding Vue that there in addition to the SFC syntax also is a Vue JSX mode. React, Solid and the new Vue support thus internally are quite similar in that they all use JSX, and the .vue SFC support is then layered on top.

Svelte only support .svelte SFC, and is thus more cumbersome to add support for, because many of the 1000+ tests then require dedicated .svelte files often with just a one-line template, that with JSX just could be trivially inlined.

**Reply by conoroshea1996** [reply]
Has a port for svelte been started so far?
Sounds like alot of work. Is the biggest hurdle the above just simply that fact there needs to be . svelte files created for each test??
It really would be great to see I'm sure the svelte community would be excited for it

**Comment by y7ya-com** [comment]
I wanted this to exist because it's easier to onboard new devs on Svelte than React, so I built an experimental port ([y7ya-com/router#feat/svelte-router](https://github.com/y7ya-com/router/tree/feat/svelte-router)) which I'm currently using in simple low-stakes projects, aiming to migrate our bigger repos from React to Svelte at some point to further test this. More details on installation & my solution for the tests problem can be found [here](https://y7ya.dev/tsr-svelte). I don't think this is necessarily the best solution to be included into the core project, but hoping I can keep it in a state where if an official port is ever finalised the migration would be as simple as changing the dependency in package.json.

**Reply by schiller-manuel** [reply]
@y7ya-com do you want to help porting router to Svelte? DM me on discord! https://discord.com/invite/WrRKjPJ

**Comment by mikecfisher** [comment]
Bump would love to see this make it into the main library :)
